### ISSUE TITLE
system certificate authorities file not loaded by default

### ISSUE THREAD

**Issue by henningpeters** [pypa/manylinux#53]
Our wheel building process contacts an https server, but fails with an `[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed` error.

Looking into it a bit more into detail I found that the system has a CA file at `/etc/pki/tls/certs/ca-bundle.crt`, but it's not loaded by default, probably because the Pythons expect it at a different location?

As a workaround I call `export SSL_CERT_FILE=/etc/pki/tls/certs/ca-bundle.crt`, but it would be great to wire it up such that the system CA file is used by default.

**Comment by njsmith** [comment]
Can you give a reproducible example, so that we can test it ourselves? :-)

**Comment by henningpeters** [comment]
Sure:

```
>>> import ssl
>>> ssl.get_default_verify_paths()
DefaultVerifyPaths(cafile=None, capath=None, openssl_cafile_env='SSL_CERT_FILE', openssl_cafile='/usr/local/ssl/ssl/cert.pem', openssl_capath_env='SSL_CERT_DIR', openssl_capath='/usr/local/ssl/ssl/certs')
```

But the paths don't exist:

```
[root@c0bddd8c075f /]# ls -l /usr/local/ssl/ssl/cert.pem
ls: /usr/local/ssl/ssl/cert.pem: No such file or directory
[root@c0bddd8c075f /]# ls -l /usr/local/ssl/ssl/certs
ls: /usr/local/ssl/ssl/certs: No such file or directory
```

Finally, you can see whether the CAs got loaded with:

```
>>> import ssl
>>> ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
>>> ctx.get_ca_certs()
[]
```

Or see the exception with:

```
>>> import urllib
>>> urllib.urlopen('https://google.com')
...
IOError: [Errno socket error] [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)
```

**Comment by njsmith** [comment]
It looks like the `get_ca_certs` check is not diagnostic -- on my up-to-date Debian testing I have:
```
>>> import ssl
>>> ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
>>> ctx.get_ca_certs()
[]
```
but cert checking is working:
```
>>> urllib.urlopen("https://google.com")
<addinfourl at 140090398193136 whose fp = <socket._fileobject object at 0x7f6957123bd0>>
>>> urllib.urlopen("https://self-signed.badssl.com/")
[...]
IOError: [Errno socket error] [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)
```

Also, the reason we didn't notice this before is that `pip` and friends all use `requests`, which ships its own vetted and up-to-date versions of the CA bundle... I strongly strongly recommend using `requests` instead of `urllib`. (This is even the official recommendation of the Python stdlib maintainers.)

In any case though, #54 should hopefully fix this... let's see what the tests say :-)

**Comment by ogrisel** [comment]
I merged #54. Please @henningpeters let us know if you still have the issue with the new version of the manylinux container (currently being built by travis at: https://travis-ci.org/pypa/manylinux/builds/127521893).
